### ISSUE TITLE
Ansible: Setup clears route to AWS instance-data special ip 169.254.169.254

### ISSUE THREAD

**Issue by lanoxx** [openvswitch/ovn-kubernetes#512]
It seems that the Ansible setup clears some routes during setup. Here are screenshots from two AWS EC2 instances, one taken after the Ansible playbook completed. The other taken from a newly setup instance before the Ansible playbook ran.

After OVN setup:

![image](https://user-images.githubusercontent.com/691311/48953046-c4995d80-ef44-11e8-96ab-9cf06b2a647a.png)

Before OVN setup:

![image](https://user-images.githubusercontent.com/691311/48954730-147b2300-ef4b-11e8-8c33-cdccf045219d.png)

This prevents access to the AWS EC2 special IP `169.254.169.254`. Manually adding back the missing route fixes the problem.

**Comment by lanoxx** [comment]
It seems this problem only occurs on Windows.

**Comment by aserdean** [comment]
Thanks for bringing this up.

I forgot that we have to copy the routes and properties from the interface on which we create the HNS network upon.

We used to copy only the MTU from the interface at some point.

I'll try to take care of it.

**Comment by lanoxx** [comment]
This is starting to become more of a problem, now that I am trying to enable the `--cloud-provider=aws` option in the Windows `kubelet`. This option need to reach the metadata IP, which is not reachable until I manually add the route back. I need to repeat that after each reboot.

@aserdean Did you have a change to take a look at this, or do you have some idea what the problem is?

**Comment by aserdean** [comment]
> This is starting to become more of a problem, now that I am trying to enable the `--cloud-provider=aws` option in the Windows `kubelet`. This option need to reach the metadata IP, which is not reachable until I manually add the route back. I need to repeat that after each reboot.
> 

Ouch! Are you adding a persistent route?

> @aserdean Did you have a change to take a look at this, or do you have some idea what the problem is?

The issue (as mentioned before), is that, when we create the hns network, unfortunately, we have to copy all of the old adapter properties on top of the new interface.

It's basically this issue: https://github.com/Microsoft/SDN/issues/84, but we have to keep in mind this happens also for:
`Get-NetAdapterAdvancedProperty` , `Get-NetRoute`, `Get-NetIPInterface`.Metric 

It isn't that difficult to implement it, basically we need to save the values from:
`Get-NetAdapterAdvancedProperty -InterfaceAlias <name>` 
`Get-NetRoute -InterfaceAlias <name>`
`(Get-NetIPInterface -InterfaceAlias <name>).Metric`
`(Get-NetIPInterface -InterfaceAlias <name>).Mtu`
where name is:
https://github.com/openvswitch/ovn-kubernetes/blob/master/contrib/roles/windows/kubernetes/tasks/setup_sdn.yml#L15
and apply them on top of the new created interface:
https://github.com/openvswitch/ovn-kubernetes/blob/master/contrib/roles/windows/kubernetes/tasks/setup_sdn.yml#L39
which will have the following interface name:
```
$interfaceName = "vEthernet ($($net.NetworkAdapterName))"
```

Unfortunately I haven't found some free cycles to implement and test it. Please let me know if you want to take a try at it.

**Comment by lanoxx** [comment]
> Ouch! Are you adding a persistent route?

No, the persistent route already exists, but it is not being restored after a reboot. If you look at  the two pictures I posted in the issue description, you can see that the metadata service (e.g. 169.254.169.254) is already visible under persistent routes both before and after the setup process.

After some more debugging I realized that, just as you explained too, the problem is that the existing persistent routes are bound to the old interface and need to be moved to the new interface. 

The primitive `route print` command that I used to capture the routing tables in the two screenshots above unfortunately does not show the interface to which the persistent routes are bound. But by using the following command we can get an output that includes the interface Id:

    Get-NetRoute -PolicyStore all

I have now manually deleted the persistent routes and added them back with the correct interface (using `netsh` or `New-NetRoute`) and now they are being restored after a reboot.

Now to your suggestion, when I try the following commands with the old interface name then only the first two commands produce some output, the last two commands just give an error for me:

    Get-NetAdapterAdvancedProperty -InterfaceAlias <name>
    Get-NetRoute -InterfaceAlias <name>
    (Get-NetIPInterface -InterfaceAlias <name>).Metric
    (Get-NetIPInterface -InterfaceAlias <name>).Mtu

It seems there is no longer an IP interface with the old name, so I think that the original MTU and Metric need to be captured from the old IP interface before the new host network is created.

Also some one more questions, my understanding is that `$interfaceName` should be equal to `vEthernet ({{ interface_name }})`, so why the complicated workaround using `$net.NetworkAdapterName`. Maybe I missed something.

Also the output of `Get-NetIPInterface` does not have properties `.Metric` or `.Mtu` only the following two if I am not mistaken:

    AutomaticMetric : Enabled
    InterfaceMetric : 15
    NlMtu: 1500

I am not quite sure how the AdvancedProperties can be moved.
